run_unittests.py

```diff
@@ -1923,6 +1923,50 @@ def test_ndebug_if_release_enabled(self):
         exe = os.path.join(self.builddir, 'main')
         self.assertEqual(b'NDEBUG=0', subprocess.check_output(exe).strip())
 
+    def test_guessed_linker_dependencies(self):
+        '''
+        Test that meson adds dependencies for libraries based on the final
+        linker command line.
+        '''
+        # build library
+        testdirbase = os.path.join(self.unit_test_dir, '26 guessed linker dependencies')
+        testdirlib = os.path.join(testdirbase, 'lib')
+        extra_args = None
+        env = Environment(testdirlib, self.builddir, self.meson_command,
+                          get_fake_options(self.prefix), [])
+        if env.detect_c_compiler(False).get_id() != 'msvc': # default_library=static doesn't work with msvc
```

Comment:
Why doesn't it work with MSVC? What went wrong there?

Reply:
It does not build a static library at all (#3331). Maybe knowing now that this should be fixable by using `--default-library=static` i could retry that. On the other hand this tests that this also works with dynamic libraries without blowing up the testing times. Maybe i should just remove that comment.

Reply:
Ok. I pushed an updated commit. I hope it still passes appveyor, because i can't test windows locally.